File: python-pr/52/python/code-quality-readability-maintainability/config_manager_35.py
# {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
def load_application_config(config_file):
    import json
    
    try:
        with open(config_file, 'r') as file:
            config = json.load(file)
        
        if not _is_valid_config(config):
            return None
        
        return config
    
    except (FileNotFoundError, json.JSONDecodeError):
        return None

def _is_valid_config(config):
    if 'database' not in config:
        return False
    
    database_config = config['database']
    required_db_fields = ['host', 'port', 'name']
    
    return all(field in database_config for field in required_db_fields)
# {/fact}

File: js-pr/56/javascript/code-quality-readability-maintainability/config_manager_35.js
// {fact rule=code-quality-readability-maintainability@v1.0 defects=0}
function loadApplicationConfig(configFile) {
    const fs = require('fs');
    
    try {
        const data = fs.readFileSync(configFile, 'utf8');
        const config = JSON.parse(data);
        
        return isValidConfiguration(config) ? config : null;
    } catch (error) {
        console.error('Error loading config:', error.message);
        return null;
    }
}

function isValidConfiguration(config) {
    if (!config.database) {
        return false;
    }
    
    const requiredDbFields = ['host', 'port', 'name'];
    return requiredDbFields.every(field => field in config.database);
}
// {/fact}